llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -966,6 +995,39 @@ Error MCCASBuilder::buildFragments() {
       continue;
     }
 
+    if (Asm.getContext().getObjectFileInfo()->getDwarfLineSection() == &Sec) {
+      SmallVector<char> DebugLineContent;
```

Comment:
Why the SmallVector and not a vector, and is the default size reasonable, or should we reserve more space for the inline elements?

Reply:
In general, the LLVM's programmer manual prescribes [always favouring](https://llvm.org/docs/ProgrammersManual.html#llvm-adt-smallvector-h) `SmallVector<T, 0>` over `std::vector`.

> SmallVector has grown a few other minor advantages over std::vector, causing SmallVector<Type, 0> to be preferred over std::vector<Type>. [...]

Regarding the "SmallSize" default, do we have any good metrics for what a type DebugLine section would look like for a "Small" function?

Reply:
I will be honest, I just used a `SmallVector` because I thought it was more common to use it in LLVM. I have no idea what would be more ideal

Reply:
Well, from the document Felipe linked, you were right :-)